### DISCUSSION TITLE
Data Host Storage

### DISCUSSION THREAD

**Discussion by frm** [trustfractal/protocol#16] [Q&A]
The current data host website tracker is keeping everything in memory for now. I'm planning on moving to local storage soon, but we should discuss two things:

1. The format - @shelbyd has some opinions on this already, but we didn't quite agree on anything.
2. How and where to relay the data - Chrome extensions are limited to 5MBs of data in the local storage. We could optionally add the `unlimitedStorage` permission but this is very eager and I'd prefer to avoid any potential problems that come with it. A potential solution is to relay the data somewhere (e.g. a "local" or native data host, with disk access).

None of these are strictly needed for _now_ but I'd rather start the discussion on them and see where we end up.

**Comment by shelbyd** [comment]
We need to also consider the case of multiple extensions including our SDK not duplicating all the behavior. Ideally one acts as the "primary" and others just contribute to the primary. 

Also need to think about what to do when the user uninstalls the primary extension. We should keep the data in that case, so another extension can take over as primary or a future extension can pick up where the other left off.

To support those, I'd think we need to store things on the file-system. I'm pretty sure extensions are able to do that?
